From 1ae8de3671061daa445eeefebf8a29bd1afbdf42 Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Sat, 29 Apr 2023 15:33:05 +0200
Subject: [PATCH] gmscompat: Spoof model, product and fingerprint with cust 
 prop

Allows you to change the product, device,  model and the fingerprint during the Safetynet certification

use severals properties : persist.sys.phh.safetyspoof, persist.sys.phh.safetyspoof.model, etc...

Change-Id: I765a7e2859bd6ae2ba78d0023b2998922c23829c
---
 .../internal/util/PropImitationHooks.java     | 23 +++++++++++++++----
 1 file changed, 18 insertions(+), 5 deletions(-)

diff --git a/core/java/com/android/internal/util/PropImitationHooks.java b/core/java/com/android/internal/util/PropImitationHooks.java
index d1080dbe6780..6d3ff6f050de 100644
--- a/core/java/com/android/internal/util/PropImitationHooks.java
+++ b/core/java/com/android/internal/util/PropImitationHooks.java
@@ -50,6 +50,17 @@ public class PropImitationHooks {
     private static final String PACKAGE_GPHOTOS = "com.google.android.apps.photos";
     private static final String PACKAGE_SNAPCHAT = "com.snapchat.android";
 
+    private static final boolean SAFETYSPOOF_ENABLE =
+            android.os.SystemProperties.getBoolean("persist.sys.phh.safetyspoof", false);
+    private static final String SAFETYSPOOF_MODEL =
+            android.os.SystemProperties.get("persist.sys.phh.safetyspoof.model", "Pixel XL");
+    private static final String SAFETYSPOOF_DEVICE =
+            android.os.SystemProperties.get("persist.sys.phh.safetyspoof.device", "marlin");
+    private static final String SAFETYSPOOF_PRODUCT =
+            android.os.SystemProperties.get("persist.sys.phh.safetyspoof.product", "marlin");
+    private static final String SAFETYSPOOF_FINGERPRINT=
+            android.os.SystemProperties.get("persist.sys.phh.safetyspoof.fingerprint", "google/marlin/marlin:7.1.2/NJH47F/4146041:user/release-keys");
+
     private static final Map<String, Object> sPixelXLProps = Map.of(
         "BRAND", "google",
         "MANUFACTURER", "Google",
@@ -78,12 +89,14 @@ public class PropImitationHooks {
          * Set stock fingerprint for ARCore
          * Set Pixel XL for Google Photos
          */
-        if (sCertifiedProps.length == 4 && sIsGms) {
+        if (sCertifiedProps.length == 4 && sIsGms && SAFETYSPOOF_ENABLE) {
             dlog("Spoofing build for GMS");
-            setPropValue("DEVICE", sCertifiedProps[0]);
-            setPropValue("PRODUCT", sCertifiedProps[1]);
-            setPropValue("MODEL", sCertifiedProps[2]);
-            setPropValue("FINGERPRINT", sCertifiedProps[3]);
+            setPropValue("DEVICE", SAFETYSPOOF_DEVICE);
+            setPropValue("PRODUCT", SAFETYSPOOF_PRODUCT);
+            setPropValue("MODEL", SAFETYSPOOF_MODEL);
+            setPropValue("FINGERPRINT", SAFETYSPOOF_FINGERPRINT);
+
+
         } else if (!sStockFp.isEmpty() && packageName.equals(PACKAGE_ARCORE)) {
             dlog("Setting stock fingerprint for: " + packageName);
             setPropValue("FINGERPRINT", sStockFp);
-- 
2.25.1

